bin/run-command.sh

```diff
@@ -18,10 +18,71 @@ set -euo pipefail
 # TODO: Add ability to change task IAM Role. Part 3 of multipart update https://github.com/navapbc/template-infra/issues/354#issuecomment-1693973424
 # TODO: Change to keyword arguments. Part 3 of multipart update https://github.com/navapbc/template-infra/issues/354#issuecomment-1693973424
 
-APP_NAME="$1"
-ENVIRONMENT="$2"
-COMMAND="$3"
-ENVIRONMENT_VARIABLES=${4:-""}
+ACCOUNT_ID="$(./bin/current-account-id.sh)"
+
+while :; do
+  # echo "$1 $2"
+  if [ -z ${1+x} ]; then
+    break
+  fi
+  case $1 in
+    --environment-variables)
+      if [ "$2" ]; then
+        ENVIRONMENT_VARIABLES=$2
+        shift 2
+      else
+        echo "$1 flag present without Environment Variables"
+        exit 1
+      fi
+      ;;
+    --role-name)
+      if [ -z ${ROLE_ARN+x} ]; then
+        if [ "$2" ]; then
+          ROLE_ARN="arn:aws:iam::$ACCOUNT_ID:role/$2"
+          shift 2
+        else
+          echo "$1 flag present without role name"
+          exit 1
+        fi
+      else
+        printf "Conflicting attempts to define role\n%s\n%s" "$ROLE_ARN" "$1"
+        exit 1
+      fi
+      ;;
```

Comment:
to begin with we'll only have one place that's using the role override, so we only need one option. having two options means we'll have unreachable/dead code, and unreachable code is bad to keep around since it creates a maintenance burden without adding any value. a useful way to think in general is that you should treat all code as a liability and minimize the amount of code we need to accomplish what we're trying to do.

Reply:
Sounds good. Do we think that role-arn or role-name makes more sense in this instance? Either should be pretty easy to get programmatically, role-name is slightly more human friendly. ARN's are a bit more "standard"

Reply:
let's go with role-arn